handlers/registration.js

```diff
@@ -27,70 +27,67 @@ module.exports.create = async (event, ctx, callback) => {
     return;
   }
   const id = parseInt(data.id, 10);
+  const eventID = data.eventID;
+  let registrationStatus = data.registrationStatus;
 
-  const params = {
-      Item: {
-          id: id,
-          eventID: data.eventID,
-          status: data.status,
-          createdAt: timestamp,
-          updatedAt: timestamp
-      },
-      TableName: 'biztechRegistration' + process.env.ENVIRONMENT
-  };
-
-  // // Update Event count
-  // let updateExpression = 'set ';
-
-  // if (data.status == 'cancelled') {
-  //     updateExpression += 'registeredNum \= registeredNum - :incr,';
-  // } else {
-  //     const num = data.status + 'Num';
-  //     updateExpression +=  num + ' \= ' + num + ' \+ :incr,';
-  // }
-
-  // let expressionAttributeValues = {':incr': 1};
-
-  // // Update timestamp
-  // updateExpression += "updatedAt = :updatedAt";
-  // expressionAttributeValues[':updatedAt'] = timestamp;
-
-  // // Log the update expression
-  // console.log(updateExpression);
-
-  // const eventParams = {
-  //   Key: {
-  //     id: data.eventID
-  //   },
-  //   TableName: 'biztechEvents' + process.env.ENVIRONMENT,
-  //   ExpressionAttributeValues: expressionAttributeValues,
-  //   UpdateExpression: updateExpression,
-  //   ReturnValues:"UPDATED_NEW"
-  // };
-  await docClient.put(params).promise()
-  .then(result => {
-      const response = {
-        statusCode: 200,
-        headers: {
-          'Access-Control-Allow-Origin': '*',
-          'Access-Control-Allow-Credentials': true,
+  const eventParams = {
+    Key: { id: eventID },
+    TableName: 'biztechEvents' + process.env.ENVIRONMENT
+  }
+  // Check if the event is full
+  await docClient.get(eventParams).promise()
+    .then(async(event) => {
+      const counts = await helpers.getEventCounts(eventID)
+
+      if (counts.registeredCount >= event.Item.capac){
```

Comment:
This should only be checked if registrationStatus is registered. If it's cancelled or checkedin, it should just go through.

Reply:
Good call